jsonrpc/ws-connection/src/ws.ts

```diff
@@ -117,7 +117,10 @@ export class WsConnection implements IJsonRpcConnection {
     this.registering = true;
 
     return new Promise((resolve, reject) => {
-      const opts = !isReactNative() ? { rejectUnauthorized: !isLocalhostUrl(url) } : undefined;
+      const origin = new URLSearchParams(url).get("origin");
+      const opts = isReactNative()
+        ? { headers: { origin } }
```

Comment:
What happens if the `origin` query param is not present? Seems like it will overwrite any existing `origin` header to `undefined` in that case rather than preserving the default `origin` header.

Reply:
I've made some tests and if the query-param is not present the origin header is not overwritten 💪